exonum-java-binding/core/rust/src/runtime/cmd.rs

```diff
@@ -114,6 +125,105 @@ pub struct Run {
     jvm_args_append: Vec<String>,
 }
 
+/// EJB-specific `run-dev` command.
+///
+/// Automatically generates node configuration for one
+/// validator and runs it using provided `artifacts_path` as a directory for Java service artifacts.
+#[derive(Debug, StructOpt, Serialize, Deserialize)]
+#[structopt(rename_all = "kebab-case")]
+pub struct RunDev {
+    /// Path to the directory containing Java service artifacts.
+    #[structopt(long)]
+    artifacts_path: PathBuf,
+    /// Path to a database directory. If not provided, system temporary directory is used.
+    ///
+    /// If not provided, new database will be created for each use of the command. Database is
+    /// not deleted automatically, the user must clean up database directory themselves.
+    #[structopt(long, short = "d")]
+    db_path: Option<PathBuf>,
+    /// Path to a node configuration file. If not provided, autogenerated node configuration is
+    /// used.
+    ///
+    /// If not provided, new node configuration will be generated for each use of the command.
+    /// Defaults to <db_path>/config if <db_path> parameter is provided. Otherwise system temporary
+    /// directory is used. Config files are not deleted automatically, the user must clean up
+    /// configuration files themselves.
+    #[structopt(long, short = "c")]
+    node_config: Option<PathBuf>,
+    /// Path to log4j configuration file.
+    #[structopt(long)]
+    ejb_log_config_path: Option<PathBuf>,
+}
+
+impl RunDev {
+    /// Returns path to a directory for node configuration files.
+    ///
+    /// If <db_path> parameter is provided, returns <db_path>/config. Else returns system temporary
+    /// directory.
+    fn config_directory(&self) -> Result<PathBuf, failure::Error> {
+        if let Some(db_path) = self.db_path.clone() {
+            let mut config_dir = db_path;
+            config_dir.push("config");
+            Ok(config_dir)
+        } else {
+            Ok(tempfile::tempdir()?.into_path())
+        }
+    }
+
+    /// Automatically generates node configuration and returns a path to node configuration file.
+    ///
+    /// Does not alter existing configuration files.
+    fn generate_node_configuration(&self) -> Result<PathBuf, failure::Error> {
+        let config_directory = self.config_directory()?;
+        let node_config_path = concat_path(config_directory.clone(), "node.toml");
+
+        // Configuration files exist, skip generation.
+        if config_directory.exists() {
```

Comment:
How does existence of a config_directory (a temp dir may be created in config_directory) imply config file existence? Have you tested this?

Reply:
Error messages when node configuration file is deleted from `database_path/config` directory:
```
Error: Os { code: 2, kind: NotFound, message: "No such file or directory" }

loading config from testnet/config/node.toml
```